frontend/app/routes/$lang/_public/status/child/index.tsx

```diff
@@ -0,0 +1,383 @@
+import { getApplicationStatusService } from '~/services/application-status-service.server';
+import { getLookupService } from '~/services/lookup-service.server';
+import * as adobeAnalytics from '~/utils/adobe-analytics.client';
+import { isValidApplicationCode } from '~/utils/application-code-utils';
+import { extractDateParts, getAgeFromDateString, isPastDateString, isValidDateString } from '~/utils/date-utils';
+import { featureEnabled, getEnv } from '~/utils/env.server';
+import { useHCaptcha } from '~/utils/hcaptcha-utils';
+import { getNameByLanguage, getTypedI18nNamespaces } from '~/utils/locale-utils';
+import { getFixedT } from '~/utils/locale-utils.server';
+import { getLogger } from '~/utils/logging.server';
+import { RouteHandleData, getPathById } from '~/utils/route-utils';
+import { formatSin, isValidSin } from '~/utils/sin-utils';
+import { cn } from '~/utils/tw-utils';
+
+enum ChildHasSin {
+  No = 'no',
+  Yes = 'yes',
+}
+
+export const handle = {
+  i18nNamespaces: getTypedI18nNamespaces('status', 'gcweb'),
+  pageIdentifier: pageIds.public.status.child.index,
+  pageTitleI18nKey: 'status:child.page-title',
+} as const satisfies RouteHandleData;
+
+export async function loader({ context: { session }, params, request }: LoaderFunctionArgs) {
+  featureEnabled('status');
+  const { ENABLED_FEATURES, HCAPTCHA_SITE_KEY } = getEnv();
+
+  const csrfToken = String(session.get('csrfToken'));
+  const t = await getFixedT(request, handle.i18nNamespaces);
+
+  const hCaptchaEnabled = ENABLED_FEATURES.includes('hcaptcha');
+
+  const meta = { title: t('gcweb:meta.title.template', { title: t('status:child.page-title') }) };
+
+  return json({ csrfToken, hCaptchaEnabled, meta, siteKey: HCAPTCHA_SITE_KEY });
+}
+
+export async function action({ context: { session }, params, request }: ActionFunctionArgs) {
+  featureEnabled('status');
+  const log = getLogger('status/child/index');
+  const { CLIENT_STATUS_SUCCESS_ID, ENABLED_FEATURES } = getEnv();
+  const hCaptchaRouteHelpers = getHCaptchaRouteHelpers();
+  const t = await getFixedT(request, handle.i18nNamespaces);
+
+  const codeSchema = z.object({
+    code: z.string().trim().min(1, t('status:child.form.error-message.application-code-required')).refine(isValidApplicationCode, t('status:child.form.error-message.application-code-valid')),
+  });
+
+  const childHasSinSchema = z.object({
+    childHasSin: z.boolean({ errorMap: () => ({ message: t('status:child.form.error-message.radio-required') }) }),
+  });
+
+  const sinSchema = z.object({
+    sin: z
+      .string()
+      .trim()
+      .min(1, t('status:child.form.error-message.sin-required'))
+      .refine(isValidSin, t('status:child.form.error-message.sin-valid'))
+      .transform((sin) => formatSin(sin, '')),
+  });
+
+  const childInfoSchema = z
+    .object({
+      firstName: z.string().max(100, t('status:child.form.error-message.first-name-too-long')).optional(),
```

Comment:
I noticed that first name is optional but the form doesn't indicate that. Is this `optional()` intentional?

Reply:
I set it as optional because you need a last name no matter what. 

Also says "If the child goes by a single legal name, leave the first name field blank".

Reply:
Understood - resolving